src/download-coursier.ts

```diff
@@ -0,0 +1,75 @@
+import * as path from "path";
+import { https } from "follow-redirects";
+import { IncomingMessage } from "http";
+import * as fs from "fs";
+
+export function getCoursier(
+  extensionPath: string,
+  versionPath: string
+): Promise<string> {
+  function binPath(filename: string) {
+    return path.join(extensionPath, filename);
+  }
+  const urls = {
+    darwin: `https://github.com/coursier/coursier/releases/download/${versionPath}/cs-x86_64-apple-darwin`,
+    linux: `https://github.com/coursier/coursier/releases/download/${versionPath}/cs-x86_64-pc-linux`,
+    win32: `https://github.com/coursier/coursier/releases/download/${versionPath}/cs-x86_64-pc-win32.exe`,
+  };
+  const targets = {
+    darwin: binPath("coursier"),
+    linux: binPath("coursier"),
+    win32: binPath("coursier.exe"),
+  };
+  return downloadFile(urls[process.platform], targets[process.platform]);
```

Comment:
before the download is triggered, this should : 

1. test whether `coursier` is on the path (by shelling out to `coursier resolve --help` and checking for a 0 exit code), and just return `coursier` if it does
2. if not on the current path, test whether the file already exists (before performing an http call), and return that if it does

Reply:
should I check the version of the coursier I find on path?

Reply:
nah, the commands are pretty stable. We'll worry about it if it breaks